src/BookAnalizer/BookStatisticCalculator.java

```diff
@@ -0,0 +1,78 @@
+package BookAnalizer;
+
+import java.io.*;
+import java.util.*;
+
+public class BookStatisticCalculator {
+    public void analyzeBook(File file) {
+        try {
+            Set<String> uniqueWords = readWordsFromFile(file);
+            List<String> filteredWords = filterWords(uniqueWords);
+            Map<String, Integer> wordCount = countWordOccurrences(filteredWords);
+            List<Map.Entry<String, Integer>> sortedWordCount = sortWordCount(wordCount);
+            int uniqueWordCount = wordCount.size();
+            writeStatisticToFile(file.getName(), sortedWordCount, uniqueWordCount);
+            displayStatistic(sortedWordCount, uniqueWordCount);
+        } catch (IOException e) {
+            System.err.println("Error reading file: " + e.getMessage());
+        }
+    }
+
+    private Set<String> readWordsFromFile(File file) throws IOException {
+        Set<String> uniqueWords = new HashSet<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] words = line.toLowerCase().split("[\\s\\W]+");
+                uniqueWords.addAll(Arrays.asList(words));
+            }
+        }
+        return uniqueWords;
+    }
+
+    private List<String> filterWords(Set<String> uniqueWords) {
+        List<String> filteredWords = new ArrayList<>();
+        for (String word : uniqueWords) {
+            if (word.length() > 2) {
+                filteredWords.add(word);
+            }
+        }
+        return filteredWords;
```

Comment:
why don't to filter while all words are counted?

Reply:
Filtering words before counting their occurrences makes sense because it avoids counting short words (in this case, words less than 3 characters long) that may not be meaningful for text analysis. If occurrence counting occurred before filtering, short words would be counted in the statistics, which could lead to incorrect results or skewed data.

In this code, filtering occurs after reading words from a file, but before counting their occurrences in the text. This allows you to ensure that only words that meet the specified criteria (length greater than 2 characters) will be taken into account in the statistics.